### DISCUSSION TITLE
Consider adding nippy as a built-in library

### DISCUSSION THREAD

**Discussion by borkdude** [babashka/babashka#1309] [Ideas]
Ideas: 

- Include nippy as a built-in (see nippy branch)
- Include nippy in jet, also expose nippy functions as pod
- Make separate nippy CLI that can also expose itself as pod

One issue:  https://github.com/ptaoussanis/nippy/issues/119 
Can probably be solved using reflection config.

cc @justone

**Comment by borkdude** [comment]
See https://github.com/justone/brisk#babashka-pod-support

**Comment by davesann** [comment]
Am I correct that, as this uses a pod, the data to be written will first be serialised, then de-serialised in the pod, then nippy serialised to a file? (and roughly the reverse for read)

So, while this works, it adds a fair amount of overhead (for larger data). 

Usage for nippy format compatibility, not speed? And if you just want to serialise - something that is in process would likely be better?

**Reply by borkdude** [reply]
@davesann Correct, there will be overhead, but it's worth measuring how much this is noticeable in practice. For huge files it may be better to do it with a JVM anyway.

Another option might be to add nippy to jet: https://github.com/borkdude/jet 
Since it's already a tool to convert between formats.
